subgraphs/arrakis-finance/src/mappings/handlers/liquidityGauge.ts

```diff
@@ -9,73 +12,158 @@ import {
 } from "../../../generated/templates/LiquidityGauge/LiquidityGaugeV4";
 import {
   getOrCreateLiquidityGauge,
-  updateRewardToken,
-  updateRewardEmission,
+  removeRewardToken,
+  updateRewardEmissions,
+  addRewardToken,
 } from "../helpers/liquidityGauge";
 import { getOrCreateVault, updateVaultSnapshots } from "../helpers/vaults";
-import { _GaugeRewardToken } from "../../../generated/schema";
+import { Vault } from "../../../generated/schema";
+import { RewardTokenType } from "../../common/constants";
+import { getOrCreateRewardToken } from "../../common/getters";
 
 export function handleAddGauge(event: AddGauge): void {
-  const gaugeAddress = event.params.gauge
-  const vaultAddress = event.params.vault
-  let gauge = getOrCreateLiquidityGauge(gaugeAddress);
-  gauge.vault = vaultAddress.toHex();
+  const gaugeAddress = event.params.gauge;
+  LiquidityGaugeTemplate.create(gaugeAddress);
+
+  const vaultAddress = event.params.vault;
+  const gauge = getOrCreateLiquidityGauge(gaugeAddress);
+  gauge.vault = vaultAddress.toHexString();
   gauge.save();
 
+  const vault = Vault.load(vaultAddress.toHexString());
+  if (!vault) {
+    log.error("[handleAddGauge]vault {} doesn't exist tx {}-{}", [
+      vaultAddress.toHexString(),
+      event.transaction.hash.toHexString(),
+      event.transactionLogIndex.toString(),
+    ]);
+    return;
+  }
+
   // Liquidity gauge sets the first reward token to SPICE in constructor
-  let gaugeContract = GaugeContract.bind(gaugeAddress);
-  const rewardTokenAddress = gaugeContract.SPICE();
-  updateRewardToken(
-    gaugeAddress,
-    vaultAddress,
-    rewardTokenAddress,
-    event.block
-  );
+  const gaugeContract = GaugeContract.bind(gaugeAddress);
```

Comment:
What does the gauge do?

If you look at the number of times this event is emitted its like 80 million... https://miniscan.xyz/?network=optimism&address=0x2ff5d1da4985113f467bbbff015e76ce8ab05f29

Reply:
gauge allows users to stake vault token and claim rewards. The gauge registry is a factory contract creating gauge contracts. I don't know what is going on with miniscan, it seems the Events link is not working correctly. This is the gauge registery we are using: https://optimistic.etherscan.io/address/0x2ff5d1da4985113f467bbbff015e76ce8ab05f29#events

Reply:
Okay that looks more legit, must be something weird with the decoding in miniscan